helix-rest/src/main/java/org/apache/helix/rest/server/ServerContext.java

```diff
@@ -136,7 +128,20 @@ public RealmAwareZkClient getRealmAwareZkClient() {
               // Refresh data subscription
               _zkClientForListener.unsubscribeAll();
               _zkClientForListener.subscribeRoutingDataChanges(this, this);
+              LOG.info("ServerContext: subscribed to routing data in routing ZK at {}!", _zkAddr);
```

Comment:
Why we change the place? Could _zkClientForListener be null? It should be constructed when the ServerContext built.

Reply:
I just changed the order so that control flow is not broken if FederatedZkClient creation throws an exception.

You can imagine a scenario in which, in the previous order, if FederatedZkClient creation failed and threw an exception, this whole resubscription logic would be skipped altogether, which is not what we want.